html/components/tooltip.js

```diff
@@ -0,0 +1,118 @@
+import getElements from '../utilities/getElements';
+import {
+  isEscPressed,
+  isEnterPressed,
+  isSpacePressed,
+} from '../utilities/keypress';
+import toggleAriaExpanded from '../utilities/toggleAriaExpanded';
+
+/**
+ *  Tooltip JS
+ */
+
+function calculatePositionClass(trigger){
+  const elemX = trigger.getBoundingClientRect().left;
+  const elemY = trigger.getBoundingClientRect().top;
+  const viewportWidth = window.innerWidth;
```

Comment:
window use should be wrapped in a check to make sure window exists

Reply:
Is this true for things like `window.addEventListener` as well?

Reply:
yep, anywhere where we assume window exists